include/aos/common/tools/map.hpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_MAP_HPP_
+#define AOS_MAP_HPP_
+
+#include "aos/common/tools/array.hpp"
+#include "aos/common/tools/utils.hpp"
+
+namespace aos {
+
+/**
+ * Simple map implementation based on a non sorted array.
+ */
+template<typename Key, typename Value>
+class Map {
+public:
+    /**
+     * Returns pointer to the value for a given key. And null if a key is absent.
+     * 
+     * @param key
+     * @return const Value*
+     */
+    Value* Get(const Key& key);
```

Comment:
Can we use `At` that returns RetWithError<Value&>?

Reply:
I would leave name "Get" to have a pair with Set function. And return a pointer as it doesn't require additional variable in client code,  a bit easier to use.